ibis/backends/datafusion/__init__.py

```diff
@@ -184,8 +182,43 @@ def execute(
                 # dummy datafusion table
                 compiled = self.compile(expr, params, **kwargs)
                 frame = self._context.empty_table().select(compiled)
-            table = _to_pyarrow_table(frame)
-            return table[0][0].as_py()
+            return frame
+        else:
+            raise com.IbisError(
+                f"Cannot execute expression of type: {type(expr)}"
+            )
+
+    def to_pyarrow_batches(
```

Comment:
Could you please mark these method as experimental? Perhaps an `@experimental` decorator would we useful similar to `ibis.util.deprecated`.

I'm not convinced yet that the eagerly evaluated API is the preferable. I'm planning to put up a prototype to use specific operations for different outputs so we can compare them, similar to https://github.com/kszucs/ibis/pull/5/files